lms/djangoapps/courseware/views/views.py

```diff
@@ -225,7 +226,10 @@ def jump_to(_request, course_id, location):
     except InvalidKeyError:
         raise Http404(u"Invalid course_key or usage_key")
     try:
-        redirect_url = get_redirect_url(course_key, usage_key)
+        if usage_key.block_type == 'course' and waffle.flag_is_active(request, 'unified_course_view'):
+            redirect_url = reverse('edx.course_experience.course_home', args=[unicode(course_key)])
+        else:
+            redirect_url = get_redirect_url(course_key, usage_key)
```

Comment:
Yeah, I wasn't sure where to put this, and I was hesitant to just pass the request through. The `unified_course_view` parameter seems like a good compromise.